src/server/game/Entities/Creature/GossipDef.cpp

```diff
@@ -224,10 +224,13 @@ void PlayerMenu::SendGossipMenu(uint32 titleTextId, ObjectGuid objectGUID)
         {
             ++count;
             data << uint32(questID);
-            data << uint32(item.QuestIcon);
+            data << uint32(quest->IsAutoComplete() ? 0 : item.QuestIcon);
```

Comment:
I think this is unneccessary - 3.3.5 client only checks 2 values
if QuestIcon == 3 || QuestIcon == 4 then quest is treated as "active"
if QuestIcon != 3 && QuestIcon != 4 then quest is treated as "available"

Reply:
I have tried with values different than 0, but they cause a bug where you can accept the auto-completable quest and it goes in the quest log.

0 seems to be the only value that works here.